dask_searchcv/adaptive.py

```diff
@@ -0,0 +1,446 @@
+
+        self.history = []
+
+        super_default = dict(scoring=None, iid=True, refit=True, cv=None,
+                             error_score='raise', return_train_score=_RETURN_TRAIN_SCORE_DEFAULT,
+                             scheduler=None, cache_cv=True)
+        for k, v in super_default.items():
+            if k in kwargs and kwargs[k] != super_default[k]:
+                warnings.warn('Hyperband ignores the keyword argument {k}.')
+        super(Hyperband, self).__init__(model, **kwargs)
+
+    def fit(self, X, y, dry_run=False, **fit_kwargs):
+        """
+        This function requires an active dask.distribtued client as it uses
+        the dask.distributed's concurrent.futures API.
+
+        This function implicitly assumes that higher scores are better. Note:
+        this matters if you are using a scoring function that measures loss,
+        where lower values imply a better model.
+
+        This function works with classes with a certain API, not only
+        scikit-learn estimators. It is assumed that the class
+
+        * implements ``partial_fit(self, X, y, **kwargs)``.
+        * implements ``set_params(self, **kwargs) -> self``
+        * implements ``score(self, X, y) -> number``
+        * supports ``sklearn.base.clone(self, safe=True)``. Support for this
+          function comes down to having an function ``get_params(self)``.
+
+        """
+        import distributed
+        client = _get_client()
+
+        variables = {'val_score': -np.inf, 'model': None, 'config': None,
+                     'eta': self.eta, 'history': []}
+        shared = {}
+        for key, value in variables.items():
+            shared[key] = distributed.Variable(key)
+            shared[key].set(value)
+
+        if '_dask_hyperband_data' in client.list_datasets():
+            client.unpublish_dataset('_dask_hyperband_data')
+
+        X_train, X_val, y_train, y_val = train_test_split(X, y, test_size=0.1)
+        data = {'train': [X_train, y_train], 'val': [X_val, y_val]}
+
+        for key in data.keys():
+            data[key][0] = client.scatter(data[key][0])
+            data[key][1] = client.scatter(data[key][1])
+        client.publish_dataset(_dask_hyperband_data=data)
+
+        # now start hyperband
+        R, eta = self.R, self.eta
+        s_max = math.floor(math.log(self.R, self.eta))
+        B = (s_max + 1) * self.R
+        kwargs = []
+        worker_n_jobs = -1 if self.n_jobs == -1 else (self.n_jobs // s_max) + 1
+        for s in reversed(range(s_max + 1)):
+            n = math.ceil(B/R * eta**s / (s+1))
+            r = R * eta ** -s
+            kwargs += [{'s': s, 'n': n, 'r': r, 'dry_run': dry_run, 'eta': eta,
+                        '_prefix': f's={s}', 'shared': shared,
+                        'n_jobs': worker_n_jobs}]
+
+        delayed = joblib.delayed
+        parallel = joblib.Parallel(n_jobs=1)
+        fn = delayed(_successive_halving)
```

Comment:
Slight preference to keep these sorts of things inline.  I personally find it somewhat costly when reading code to dereference functions.